### PR TITLE
rewire metadata methods to not return dataset as a side effect, chang…

### PR DESCRIPTION
**Pull request by rbanick**
…e tests correspondingly

### CODE REVIEW

**Review by chrisrossi — APPROVED** [review]
LGTM

**Comment by rbanick** [comment]
Testing on our managers reveals that `rename_data_variables` _needs_ to return a dataset for changes to register. In effect, changes to attributes are registered on the existing `xr.Dataset` object whereas changes to data variables create a new `xr.Dataset` object we need to pass through and work on.

File: gridded_etl_tools/dataset_manager.py [hunk 1]
```diff
@@ -323,7 +323,7 @@ def set_zarr_metadata(self, dataset):  # pragma NO COVER
         Method to align Zarr metadata with requirements of Zarr exports and STAC metadata format
         Happens after `populate_metadata` and immediately before data publication.
         """
-        super().set_zarr_metadata(dataset)
+        return super().set_zarr_metadata(dataset)
```
Comment (chrisrossi) [inline gridded_etl_tools/dataset_manager.py:326]: Should return type be declared in function signature?

File: gridded_etl_tools/utils/metadata.py [hunk 2]
```diff
@@ -562,7 +563,7 @@ def populate_metadata(self):
     def set_zarr_metadata(self, dataset: xr.Dataset) -> xr.Dataset:
         """
         Function to append to or update key metadata information to the attributes and encoding of the output Zarr.
-        Additionally filters out unwanted keys and fields.
+        Additionally filters out or invalid keys and fields.
```
Comment (chrisrossi) [inline gridded_etl_tools/utils/metadata.py:566]: Typo

**Review by chrisrossi — APPROVED** [review]
Looks good with minor nits.

**Pull request merged**